src/game/physics.rs

```diff
@@ -0,0 +1,51 @@
+//! Run a very simple physics simulation.
+
+use bevy::{
+    ecs::component::{ComponentHooks, StorageType},
+    prelude::*,
+};
+
+pub(super) fn plugin(app: &mut App) {
+    // `FixedUpdate` runs before `Update`, so the physics simulation is advanced
+    // before the player's visual representation is updated.
+    app.add_systems(FixedUpdate, advance_physics);
+}
+
+/// How many units per second the player should move.
+#[derive(Debug, Component, Clone, Copy, PartialEq, Default, Deref, DerefMut)]
+pub(crate) struct Velocity(pub(crate) Vec3);
+
+/// The actual transform of the player in the physics simulation.
+/// This is separate from the `Transform`, which is merely a visual
+/// representation.
+
+#[derive(Debug, Clone, Copy, PartialEq, Default, Deref, DerefMut)]
+pub(crate) struct PhysicalTransform(pub(crate) Transform);
+
+/// Make sure that this component is always initialized
+/// with the same value as the `Transform`
+impl Component for PhysicalTransform {
+    const STORAGE_TYPE: StorageType = StorageType::Table;
+
+    fn register_component_hooks(hooks: &mut ComponentHooks) {
+        hooks.on_add(|mut world, entity, _component_id| {
+            let rendered_transform = *world.get::<Transform>(entity).unwrap();
+            let mut physical_transform = world.get_mut::<PhysicalTransform>(entity).unwrap();
+            physical_transform.0 = rendered_transform;
+        });
+    }
+}
```

Comment:
This is basically the entity version of `FromWorld`, maybe `FromWorldEntity` trait is something that we might want in Bevy

Reply:
That would have a weird overlap with lifecycle hooks now though. Except maybe as a convenience trait that implemented hooks for you in the background?

Reply:
/shrug just an idea, I feel like we'll be seeing this pattern a lot

Reply:
Yeah, Avian physics already has it in place in a bunch of types. Let's see what ergonomics we develop in the future, hooks are still brand new after all :)

Reply:
https://discord.com/channels/691052431525675048/749335865876021248/1259901488428224554